lua/lualine/components/diagnostics/init.lua

```diff
@@ -104,13 +104,18 @@ function M:update_status()
 
   -- format the counts with symbols and highlights
   if self.options.colored then
-    local colors = {}
+    local colors, bgs = {}, {}
     for name, hl in pairs(self.highlight_groups) do
       colors[name] = modules.highlight.component_format_highlight(hl)
+      bgs[name] = self.options.diagnostics_color[name].bg or nil
```

Comment:
color options can be string & function too. In that case it'll through error . Get the bg from the hl_group_name in colors[name] with extract_highlight_color function from utils module .